### PR TITLE
automatically use compact formatting for pie chart total

### LINKED ISSUE #26916: Pie chart does not display does not allow custom configuration of the display of values (compact values)
**Is your feature request related to a problem? Please describe.**
When the chart is embedded together with others in a dashboard and has very large values, the visualization is impaired because the value of the series is below the chart.

**Describe the solution you'd like**
I think that if there was the possibility of configuring the display of the series value between automatic, compact and full (as some allow) or the font automatically reduced the visualization it would not be so bad

**Describe alternatives you've considered**
I increased the embedded chart card size, but that wasn't an elegant solution.

**How important is this feature to you?**
I consider it important because the visualization of information in an elegant way brings value to the use of the solution.

**Additional context**
This is a print of how it is now.
image

It could be displayed as 16.6M or the font automatically adjusted by reducing its size. But I think shrinking the font might not be the best solution.

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Closes https://github.com/metabase/metabase/issues/26916

### Description

Will automatically use compact number formatting for the total/slice value text in the center of the pie chart, if the text would overflow without it.

### How to verify

Create the following question, visualize as a pie chart, add to a dashboard and make the card small

Screenshot 2024-08-05 at 2 16 43 PM

### Demo

https://github.com/user-attachments/assets/7cb47efb-f99a-4c59-abaf-a0622574912b

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by EmmadUsmani** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/46505?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#46505** <a href="https://app.graphite.dev/github/pr/metabase/metabase/46505?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#46466** <a href="https://app.graphite.dev/github/pr/metabase/metabase/46466?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EmmadUsmani and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by EmmadUsmani** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/46505?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#46505** <a href="https://app.graphite.dev/github/pr/metabase/metabase/46505?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#46466** <a href="https://app.graphite.dev/github/pr/metabase/metabase/46466?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EmmadUsmani and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by cdeweyx — APPROVED** [review]

File: .loki/reference/chrome_laptop_Visualizations_ChartSkeleton_Pie.png
Comment (alxnddr) [inline .loki/reference/chrome_laptop_Visualizations_ChartSkeleton_Pie.png:1]: Nice to have but I wonder where this is coming from

**Review by alxnddr — APPROVED** [review]

File: .loki/reference/chrome_laptop_Visualizations_ChartSkeleton_Pie.png
Comment (EmmadUsmani) [reply .loki/reference/chrome_laptop_Visualizations_ChartSkeleton_Pie.png:1]: Not sure, I thought this was already on master, maybe it wasn't so the snapshot got created

**Pull request merged**